inference/core/workflows/core_steps/fusion/detections_stitch/v1.py

```diff
@@ -0,0 +1,252 @@
+        ]
+    ) = Field(
+        description="The output of a detection model describing the bounding boxes to be merged.",
+        examples=["$steps.my_object_detection_model.predictions"],
+    )
+    overlap_filtering_strategy: Union[
+        Literal["none", "nms", "nmm"],
+        WorkflowParameterSelector(kind=[STRING_KIND]),
+    ] = Field(
+        default="nms",
+        description="Which strategy to employ when filtering overlapping boxes. "
+        "None does nothing, NMS discards surplus detections, NMM merges them.",
+        examples=["nms", "$inputs.overlap_filtering_strategy"],
+    )
+    iou_threshold: Union[
+        FloatZeroToOne,
+        WorkflowParameterSelector(kind=[FLOAT_ZERO_TO_ONE_KIND]),
+    ] = Field(
+        default=0.3,
+        description="Parameter of overlap filtering strategy. If box intersection over union is above this "
+        " ratio, discard or merge the lower confidence box.",
+        examples=[0.4, "$inputs.iou_threshold"],
+    )
+
+    @classmethod
+    def get_dimensionality_reference_property(cls) -> Optional[str]:
+        return "reference_image"
+
+    @classmethod
+    def get_input_dimensionality_offsets(cls) -> Dict[str, int]:
+        return {"predictions": 1}
+
+    @classmethod
+    def describe_outputs(cls) -> List[OutputDefinition]:
+        return [
+            OutputDefinition(
+                name="predictions",
+                kind=[
+                    BATCH_OF_OBJECT_DETECTION_PREDICTION_KIND,
+                    BATCH_OF_INSTANCE_SEGMENTATION_PREDICTION_KIND,
+                ],
+            ),
+        ]
+
+    @classmethod
+    def get_execution_engine_compatibility(cls) -> Optional[str]:
+        return ">=1.0.0,<2.0.0"
+
+
+class DetectionsStitchBlockV1(WorkflowBlock):
+
+    @classmethod
+    def get_manifest(cls) -> Type[WorkflowBlockManifest]:
+        return BlockManifest
+
+    def run(
+        self,
+        reference_image: WorkflowImageData,
+        predictions: Batch[sv.Detections],
+        overlap_filtering_strategy: Optional[Literal["none", "nms", "nmm"]],
+        iou_threshold: Optional[float],
+    ) -> BlockResult:
+        re_aligned_predictions = []
+        for detections in predictions:
+            detections_copy = deepcopy(detections)
+            resolution_wh = retrieve_crop_wh(detections=detections_copy)
+            offset = retrieve_crop_offset(detections=detections_copy)
+            detections_copy = manage_crops_metadata(
+                detections=detections_copy,
+                offset=offset,
+                parent_id=reference_image.parent_metadata.parent_id,
+            )
+            re_aligned_detections = move_detections(
+                detections=detections_copy,
+                offset=offset,
+                resolution_wh=resolution_wh,
+            )
+            re_aligned_predictions.append(re_aligned_detections)
+        overlap_filter = choose_overlap_filter_strategy(
```

Comment:
should we be merging / detecting duplicates each time we add boxes rather than once at the end?  Maybe that doesn't make a difference?

Reply:
we should do it at the end, otherwise we are running unneeded computations I am afraid, besides - this is the implementation of `sv.InferenceSlicer` that I hope will become part of public interface later on, such that we could replace our duplicate code with sv